JeMPI_Apps/JeMPI_UI/src/hooks/useAuth.tsx

```diff
@@ -120,36 +80,116 @@ export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
   useEffect(
     () => {
       if (config.useSso) {
-        if (!isLoading) {
-          if (!user && !isLoginPage) {
+        if (!authContext.isLoading) {
+          if (!authContext.user && !isLoginPage) {
             navigate({ pathname: '/login' })
-          } else if (user && isLoginPage) {
+          } else if (authContext.user && isLoginPage) {
```

Comment:
Too many `if` conditions, this `useEffect` could be refactored to more cleaner version.